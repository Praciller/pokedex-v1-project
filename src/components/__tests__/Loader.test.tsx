import React from 'react';
import { render, screen } from '@testing-library/react';
import Loader from '../Loader';

describe('Loader component', () => {
  it('should render loader image', () => {
    render(<Loader />);
    
    const loaderImage = screen.getByRole('img');
    expect(loaderImage).toBeInTheDocument();
    expect(loaderImage).toHaveAttribute('alt', '');
  });

  it('should have correct CSS class', () => {
    const { container } = render(<Loader />);
    
    const loaderDiv = container.querySelector('.loader');
    expect(loaderDiv).toBeInTheDocument();
  });

  it('should render pokeball loader gif', () => {
    render(<Loader />);
    
    const loaderImage = screen.getByRole('img');
    expect(loaderImage.getAttribute('src')).toContain('pokeball-loader.gif');
  });

  it('should match snapshot', () => {
    const { container } = render(<Loader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
