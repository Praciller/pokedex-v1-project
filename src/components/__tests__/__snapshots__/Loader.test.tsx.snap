// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match snapshot 1`] = `
<div
  class="loader"
>
  <img
    alt=""
    src="pokeball-loader.gif"
  />
</div>
`;
